File: rec-sepval/src/main/java/net/kimleo/rec/sepval/SepValEntry.java
package net.kimleo.rec.sepval;

import net.kimleo.rec.concept.Indexible;

import java.util.List;

public class SepValEntry implements Indexible<String> {

    private final List<String> values;
    private final String source;

    public SepValEntry(List<String> values, String source) {
        this.values = values;
        this.source = source;
    }

    @Override
    public String get(int index) {
        return values.get(index);
    }

    @Override
    public int size() {
        return values.size();
    }

    public List<String> getValues() {
        return values;
    }

    public String getSource() {
        return source;
    }
}

File: rec-sepval/src/main/kotlin/net/kimleo/rec/sepval/SepValEntry.kt
package net.kimleo.rec.sepval

import net.kimleo.rec.concept.Indexible

data class SepValEntry(val values: List<String>, val source: String): Indexible<String> {

    override val size = values.size

    override operator fun get(index: Int) = values.get(index)
}